### PR TITLE
Add Implement 2 to master

### PR DESCRIPTION
**Pull request by pinterid**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] New feature (non-breaking change which adds functionality)
 - [x] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently the basic setup is on master. #6 
- Currently the README is on master. #7 
- Currently Intel 1 is on master. #12 

**What is the new behavior (if this is a feature change)?**
- Add UI 1 (Assignees: @Aichnerc ) #43 
  + Behaviour before Pull Request:
There is no usable UI. #13 
  + Behaviour after Pull Request:
Some components have been added (basic structure). #13 
- Add GitHubUtils 1 (Assignees: @auernige ) #44 
  + Behaviour before Pull Request:
Currently there is no way to store data from GitHub. #15 
Currently there is no way to get data from GitHub. #16 
  + Behaviour after Pull Request:
Now data can be stored in an alasql database. #15 
Now data can be fetched via web scraping. #16 
- Add Contributing 1 (Assignees: @kleberbaum ) #45 
  + Behaviour before Pull Request:
Currently we use the CONTRIBUTING of @aichner. #27 
  + Behaviour after Pull Request:
Now the CONTRIBUTING is adjusted for this repository. #27 
- Add COC 1 (Assignees: @kleberbaum ) #46 
  + Behaviour before Pull Request:
Currently we use the CODE_OF_CONDUCT of @aichner. #26 
  + Behaviour after Pull Request:
Now the CODE_OF_CONDUCT is adjusted for this repository. #26 
- Add UI 2 (Assignees: @Aichnerc ) #48 
  + Behaviour before Pull Request:
Currently there is no profile. #20 
Currently there is no 3D chart. #21 
  + Behaviour after Pull Request:
Now a profile page has been added. #20 
Now a 3D chart has been added. #21 
- Add GitHubUtils 2 (Assignees: @auernige ) #50 
  + Behaviour before Pull Request:
Currently the translator fails to handle the data. #23 
  + Behaviour after Pull Request:
Now the translator is able to handle the data. #23 
- Add GitLabUtils 1 (Assignees: @UnknownExecute ) #51 
  + Behaviour before Pull Request:
Currently there is no way to get data from GitLab. #30 
Currently there is no way to store data from GitLab. #31 
  + Behaviour after Pull Request:
Now data can be fetched via web scraping. #30 
Now data can be stored in an alasql database. #31 
- Add OAuth 1 (Assignees: @pinterid ) #52 
  + Behaviour before Pull Request:
Currently GitHub requests are authorized with an private access token. #33 
  + Behaviour after Pull Request:
Now GitHub requests are authorized with an OAuth2 acces token. #33 
- Improve Intel 1 (Assignees: @schettn ) #56 
  + Behaviour before Pull Request:
Currently there is no repoUrl in the repository table. #18 
Currently there is no way to create repositories without relations. #19 
Currently it is not possible to get data of user, repositories, organisations, languages, calendar, statistic and contribution types from the database. #53 
Currently the generator weights a bit off. #54 
Currently the busiest day is stored in a database table. #55 
  + Behaviour after Pull Request:
Now there is a repoUrl in the repository table. #18 
Now there is a way to create repositories without relations. #19 
Now it is possible to get data of user, repositories, organisations, languages, calendar, statistic and contribution types from the database. #53 
Now the generator weights has been adjusted. #54 
Now the busiest day is calculated from calendar days. #55 
- Add UI 3 (Assignees: @Aichnerc ) #58 
  + Behaviour before Pull Request:
Currently there is no Register component and no landing page. #35 
  + Behaviour after Pull Request:
Now a Landing page and Register component has been added. #35 
The MDB React Pro version has been updated.
- Add GitHubUtils 3 (Assignees: @kleberbaum, @schettn, @auernige ) #59   
  + Behaviour before Pull Request:
Currently there is no good way to store issues and pullRequests. #49 
Currently the contribs table gets filled with many duplicates. #41 
Currently the translator can't handle the data from Github. #23 
  + Behaviour after Pull Request:
Now all issues and pull requests get stored in the database. #49 
Now the contribs table gets filled with the correct count of contributions. #41 
Now the translator can handle all data from Githubs GraphQL query. #23 
- Add UI 4 (Assignees: @Aichnerc ) #60 
  + Behaviour before Pull Request:
Currently, the background of the Landing page is not fitting the theme. #37 
Currently, when choosing usernames, they can be duplicate. #38 
Currently, there are no popovers and tool-tips.
  + Behaviour after Pull Request:
The background of the Landing page has been updated. #37 
Duplicates shall no longer happen when choosing usernames. #38 
Popovers and tool-tips have been added.
- Add OAuth 2 (Assignees: @pinterid ) #62 
  + Behaviour before Pull Request:
OAuth is storing the data in the local storage #61 
  + Behaviour after Pull Request:
OAuth function is now returning the access token and the username #61 
- Add Intel 2 (Assignees: @schettn ) #65 
  + Behaviour before Pull Request:
Currently there is no current calendar year. #63 
Currently the calendar is generated in a wrong way. #64 
  + Behaviour after Pull Request:
Now there is a current calendar year. #63 
Now the calendar is generated in the right way. #64 
- Add GitLabUtils 2 (Assignees: @schettn ) #68 
  + Behaviour before Pull Request:
Currently the date of all calendar days are incorrect. #67 
  + Behaviour after Pull Request:
Now the date of all calendar days are correct. #67 
- Add UI 5 (Assignees: @Aichnerc ) #74 
  + Behaviour before Pull Request:
Currently there is only a "Sing up" card on the landing page. #69 
Currently the landing page is boring as hell. #70 
Currently there is no search bar. #71 
Currently there is no way to trigger the register graphql mutation of our engine. #72 
Currently there is no way to trigger the login graphql mutation of our engine. #73 
  + Behaviour after Pull Request:
Now there is a way to switch between "sing-up" and a "sing-in" on the landing page. #69 
Now the landing page is fire. #70 
Now there is a search bar. #71 
Now there is a way to trigger the register graphql mutation of our engine. #72 
Now there is a way to trigger the login graphql mutation of our engine. #73 
- Add Intel 3 (Assignees: @Aichnerc ) #77 
  + Behaviour before Pull Request:
The current calendar year is generated in a wrong way. #75 
Currently there is no start and end date of the current calendar year. #76 
  + Behaviour after Pull Request:
Now the current calendar year is generated correctly. #75 
Now there is a start and end date of the current calendar year. #76 
- Add UI 7 (Assignees: @Aichnerc ) #79 
  + Behaviour before Pull Request:
Currently, there is no contribution chart. #78 
  + Behaviour after Pull Request:
A contribution chart has been added. #78 
- Add UI 8 (Assignees: @pinterid ) #81 
  + Behaviour before Pull Request:
Currently GitLab accounts can not be linked to SNEK accounts #80 
  + Behaviour after Pull Request:
GitLab accounts can now be linked to a SNEK account.
- Add UI 6 (Assignees: @ ) #82 
  + Behaviour before Pull Request:
Currently the login is not implemented. #83 
  + Behaviour after Pull Request:
The login has been implemented. #83 
Moment.js has been added to help manage JS dates.
- Add UI 9 (Assignees: @Aichnerc ) #85 
  + Behaviour before Pull Request:
Currently, there is no footer and about link. #84 
  + Behaviour after Pull Request:
A footer including an about link has been added. #84 
- Improve Intel 4 (Assignees: @schettn ) #91 
  + Behaviour before Pull Request:
Currently there is no basic exception handling present in intel. #86 
Currently the user query is only send once when an error occurs. #87 
Currently there is one missing week in all calendars. #88 
Currently the calendar days are generated with incorrect start end end dates. #89 
Currently the contribution types lead to incorrect total values. #90 
  + Behaviour after Pull Request:
Now there is basic exception handling in intel. #86 
Now it is possible to get user data even when an error occurs the first few times. #87 
Now the amount of calendar weeks is correct. #88 
Now the calendar year range has the correct start and end date. #89 
Now the contribution types are removed to ensure correct calendar data. #90 
- Add UI 10 (Assignees: @pinterid ) #94 
  + Behaviour before Pull Request:
Currently there is no response after login. #92 
Currently there is no response after registration. #93 
  + Behaviour after Pull Request:
Now there is an notification after the login process. #92 
Now there is an notification after the register process. #93 
- Add UI 11 (Assignees: @schettn ) #97 
  + Behaviour before Pull Request:
Currently there is an empty contribution type chart on the profile page. #96 
  + Behaviour after Pull Request:
Now there is no empty contribution type chart on the profile page. #96 

**Other information**:
- **Documentation:**
  + Ref: https://www.overleaf.com/read/bcxwhwbhrmps
- **References of all involved pull requests:**
  + Ref: #42 The quality of the branch implement-1 is not sufficient.
  + Add GitHubUtils 1 (Assignees: @auernige ) #44 
    * Ref: https://github.com/agershun/alasql
    * Ref: https://github.com/snek-at/concept/blob/master/src/utils/GitHubUtils/index.js
    * Ref: https://github.com/snek-at/concept/blob/master/src/handler/GitHub/index.js
  + Add Contributing 1 (Assignees: @kleberbaum ) #45 
    * Ref: https://github.com/torvalds/linux/blob/master/Documentation/process/code-of-conduct.rst
  + Add COC 1 (Assignees: @kleberbaum ) #46 
    * Ref: https://github.com/torvalds/linux/blob/master/Documentation/process/howto.rst
  + Add GitLabUtils 1 (Assignees: @UnknownExecute ) #51 
    * Ref: https://github.com/agershun/alasql
    * Ref: https://github.com/snek-at/concept/blob/master/src/utils/GitLabUtils/index.js
    * Ref: https://github.com/snek-at/concept/blob/master/src/handler/GitLab/index.js
  + Add OAuth 1 (Assignees: @pinterid ) #52 
    * Ref: https://www.overleaf.com/read/wmvqyxgjgsyj
    * Ref: https://developer.github.com/apps/building-oauth-apps/authorizing-oauth-apps/
    * Ref: https://medium.com/@mattmazzola/react-simple-auth-react-redux-oauth-2-0-de6ea9df0a63
    * Ref: https://github.com/mattmazzola/react-simple-auth
    * Ref: https://dhavalkapil.com/blogs/Attacking-the-OAuth-Protocol/
    * Ref: https://yeti.co/blog/client-vs-server-oauth-flows-with-rest-apis/
  + Improve Intel 1 (Assignees: @schettn ) #56 
    * Ref: https://github.com/agershun/alasql
  + Add OAuth 2 (Assignees: @pinterid ) #62 
    * Ref: https://developer.github.com/apps/building-oauth-apps/authorizing-oauth-apps/
  + Add UI 5 (Assignees: @Aichnerc ) #74 
    * Ref: https://www.npmjs.com/package/react-typist
    * Ref: https://www.npmjs.com/package/react-particles-js
    * Ref: https://graphql.org/learn/queries/#mutations
    * Ref: https://github.com/snek-at/engine
    * Ref: https://jwt.io/introduction
  + Add UI 10 (Assignees: @pinterid ) #94 
    * Ref: https://mdbootstrap.com/docs/jquery/javascript/notifications/
  + Add UI 11 (Assignees: @schettn ) #97 
    * Ref: https://developer.github.com/v3/
    * Ref: https://developer.github.com/v4/
- **Dependences:**
  + AlaSQL #99 
  + Apollo #100 
  + GraphQL #101 
  + JS-SHA256 #102 
  + Moment #103 
  + React #104 
  + TypeScript #105 
  + React-Very-Simple-OAuth #106 
- **Troubleshooting:**
  + **TSID1** - Global storage @ rendering of dashboard (Profile\index.jsx)
  + **TSID2** - Missing token payload @ Token verification with snek server (App.js)
  + **TSID3** - GraphQL mutation error @ Verify snek JWT (App.js)
  + **TSID4** - GraphQL mutation error @ Login snek JWT (App.js)
  + **TSID5** - GraphQL mutation error @ Refreshing snek JWT every 2min (App.js)
  + **TSID6** - Database error message @ Saving generated user data (App.js)

### CODE REVIEW

File: src/intel/Database/Statements/Select/index.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+  calendar
+WHERE
+  total in (
+    SELECT
+      max(c2.total) as total,
+      YEAR(date)
+    FROM calendar c2
+    GROUP BY YEAR(date)
+  )
+`;
```
Comment (Aichnerc) [inline src/intel/Database/Statements/Select/index.js]: The SPDX-License is missing.

File: src/intel/utils/UtilsGithub/index.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
+import * as apollo from "./../UtilsApollo";
+import * as gqlData from "./GqlData";
+import * as translator from "./Translator";
```
Comment (Aichnerc) [inline src/intel/utils/UtilsGithub/index.js]: Please add information about the functions / contents of this files.

File: src/intel/utils/UtilsGitlab/index.js [hunk 3]
```diff
@@ -0,0 +1,244 @@
+import * as insert from "../../Database/Statements/Insert";
+import * as update from "../../Database/Statements/Update";
+import * as webscrap from "../UtilsWebscrap";
```
Comment (Aichnerc) [inline src/intel/utils/UtilsGitlab/index.js]: Please add information about the functions / contents of this files.

**Comment by pinterid** [comment]
> Devrait être d'accord pour fusionner. (●'◡'●)

נחמד

**Comment by pinterid** [comment]
Can we add a deploy script? @kleberbaum

File: src/intel/utils/UtilsGithub/Translator/index.js [hunk 4]
```diff
@@ -61,7 +61,7 @@ const getYearsDict = days => {
   let years = {};
   days.forEach(day => {
     const year = new Date(day.date).getFullYear();
-    if (years[parseInt(year)] === null) {
```
Comment (gamgamstyle) [inline src/intel/utils/UtilsGithub/Translator/index.js]: Check parseInt helix

Comment (Aichnerc) [reply src/intel/utils/UtilsGithub/Translator/index.js]: Check syntax `parseInt(string, radix);`

The radix should be included.

Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt

_Please include references and detailed information including code snippets whenever possible._

**Pull request merged**